lib/study.py

```diff
@@ -0,0 +1,248 @@
+                'fc.3': 'fort.3.mad',
+                'fc.3.aux': 'fort.3.aux',
+                'fc.8': 'fort.8',
+                'fc.16': 'fort.16',
+                'fc.34': 'fort.34'}
+        self.oneturn_sixtrack_params = {
+                "turnss": 1,
+                "nss": 1,
+                "ax0s": 0.1,
+                "ax1s": 0.1,
+                "imc": 1,
+                "iclo6": 2,
+                "writebins": 1,
+                "ratios": 1,
+                "Runnam": 'FirstTurn',
+                "idfor": 0,
+                "ibtype": 0,
+                "ition": 0,
+                "CHRO": '/',
+                "TUNE": '/',
+                "POST": 'POST',
+                "POS1": '',
+                "ndafi": 1,
+                "tunex": 62.28,
+                "tuney": 60.31,
+                "inttunex": 62.28,
+                "inttuney": 60.31,
+                "DIFF": '/DIFF',
+                "DIF1": '/',
+                "pmass": 938.272013,
+                "emit_beam": 3.75,
+                "e0": 7000,
+                "bunch_charge": 1.15E11,
+                "CHROM": 0,
+                "chrom_eps": 0.000001,
+                "dp1": 0.000001,
+                "dp2": 0.000001,
+                "chromx": 2,
+                "chromy": 2,
+                "TUNEVAL": '/',
+                "CHROVAL": '/'}
+        self.oneturn_sixtrack_input['input'] = self.madx_output
+        self.oneturn_sixtrack_output = ['fort.10']
+        self.sixtrack_output = ['fort.10']
+
+        self.tables = {
+                "mad6t_run": ['seed', 'mad_in', 'mad_out', 'fort.2', 'fort.3',\
+                        'fort.8', 'fort.16', 'job_stdout', 'job_stderr',\
+                        'job_stdlog', 'mad_out_mtime'],
+                "six_input": list(self.sixtrack_params.keys()) + ['id_mad6t_run'],
+                "six_beta": ['seed', 'tunex', 'tuney', 'beta11', 'beta12',\
+                        'beta22', 'beta21', 'qx', 'qy', 'id_mad6t_run'],
+                "dymap_results": []}
+
+    def submit_mad6t(self, platform = 'local', **args):
+        '''Submit the jobs to cluster or run locally'''
+        clean = False
+        if platform == 'local':
+            if 'place' in args:
+                execution_field = args['place']
+            else:
+                execution_field = 'temp'
+            execution_field = os.path.abspath(execution_field)
+            if not os.path.isdir(execution_field):
+                os.makedirs(execution_field)
+            if os.listdir(execution_field):
+                clean = False
+                print("Caution! The folder %s is not empty!"%execution_field)
+            cur_path = os.getcwd()
+            os.chdir(execution_field)
+            if 'clean' in args:
+                clean = args['clean']
+            for i in self.mad6t_joblist:
+                print("The job %s is running...."%i)
+                mad6t_oneturn.run(i)
+            print("All jobs are completed normally!")
```

Comment:
mmm I guess we should put in place some error handling here...

Reply:
Actually I did before, I added a ```try-except``` statement here. But I found when the job crash, I couldn't get the specific errors with ```try-except``` statement, only the ```except``` info was printed. So I removed the ```try-except``` statement. But we do need some checks before submitting, I will keep this in mind and try to add some checks here.

Reply:
or maybe we use some error-reporting in `mad6t_oneturn.run`...